test/OpenFeature.Tests/OpenFeatureTests.cs

```diff
@@ -103,8 +103,8 @@ public void OpenFeature_Should_Not_Change_Named_Providers_When_Setting_Default_P
         {
             var openFeature = Api.Instance;
 
-            openFeature.SetProvider(new NoOpFeatureProvider());
-            openFeature.SetProvider(TestProvider.DefaultName, new TestProvider());
+            openFeature.SetProviderAsync(new NoOpFeatureProvider());
+            openFeature.SetProviderAsync(TestProvider.DefaultName, new TestProvider());

```

Comment:
I think these tests should probably be made async. I've been seeing some flaky test failures and pet theory is un-awaited tasks are the culprit.

- [ ] `OpenFeature_Should_Not_Change_Named_Providers_When_Setting_Default_Provider`
- [ ] `OpenFeature_Should_Set_Default_Provide_When_No_Name_Provided`
- [ ] `OpenFeature_Should_Assign_Provider_To_Existing_Client`
- [ ] `OpenFeature_Should_Allow_Multiple_Client_Names_Of_Same_Instance`
- [ ] `OpenFeature_Should_Get_Metadata`
- [ ] `OpenFeature_Should_Allow_Multiple_Client_Mapping`

Reply:
I also saw some flaky tests, but I think they are all to do with the new events functionality, and potentially related to the fact we have some handlers we don't clean up.

Reply:
I will make these async though, there's no reason not to, you're right. They probably should have been when `SetProvider` was made async.

I _think_ the test flakiness is coming from elsewhere as I said above, but let's see. :+1:

Reply:
@austindrenski you may have been right about the test flakiness being due to missed awaits in tests after we made `SetProvider` async. I've run the tests a few times and they seem stable so far :sunglasses: .

Reply:
🍻🍻🍻🍻

Reply:
ugh nope, I got it again. I'm fairly confident I've actually fixed it here: https://github.com/open-feature/dotnet-sdk/pull/194